### DISCUSSION TITLE
`quire preview` and `quire site` fails on new project

### DISCUSSION THREAD

**Discussion by stamen-ag** [thegetty/quire#453] [Get Help]
I'm trying to create a demo version of quire, but `quire preview`, `quire site`, and other commands fail on the project generated by `quire new`. 

First off: I installed latest version, `v0.20.4`

```
$ npm install -g @thegetty/quire-cli
$ quire --version
v0.20.4
```

Then I created a new quire project:

```
$ quire new quire-demo
$ cd quire-demo/
```

Now, running `quire preview` fails.

```
$ quire preview
Launching preview server
ℹ Navigate to http://localhost:1313 to see your changes.
ℹ Press Control+C or type "quire stop" to stop the preview.
node:events:505
      throw er; // Unhandled 'error' event
      ^

Error: spawn node_modules/.bin/webpack ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn node_modules/.bin/webpack',
  path: 'node_modules/.bin/webpack',
  spawnargs: [
    '--watch',
    '-d',
    '--progress',
    '--color',
    '--config',
    'webpack/webpack.config.dev.js'
  ]
}
```

One odd thing: there is no `node_modules` or even `package.json` in the generated project directory. 

```
$ ls
 bin            config       content   README.md   static
 CHANGELOG.md   config.yml   data      resources   themes
```

<details>
<summary>show all files</summary>
<pre>
$ tree .
.
├── CHANGELOG.md
├── README.md
├── bin
│   └── deploy.sh
├── config
│   ├── _default
│   │   └── config.yml
│   ├── development
│   │   └── config.yml
│   ├── environments
│   │   ├── dev.yml
│   │   ├── github.yml
│   │   └── netlify.yml
│   ├── epub.yml
│   ├── pdf.yml
│   └── site.yml
├── config.yml
├── content
│   ├── about.md
│   ├── bibliography.md
│   ├── catalogue
│   │   ├── 1.md
│   │   ├── 2.md
│   │   ├── 3.md
│   │   ├── 4.md
│   │   └── catalogue-index.md
│   ├── contents.md
│   ├── contributors.md
│   ├── cover.md
│   ├── essay.md
│   ├── intro.md
│   └── search-index.md
├── data
│   ├── figures.yml
│   ├── objects.yml
│   ├── publication.yml
│   └── references.yml
├── resources
│   └── _gen
│       ├── assets
│       └── images
├── static
│   ├── css
│   │   └── custom.css
│   ├── data
│   │   └── sample-geojson.json
│   ├── downloads
│   │   ├── output.epub
│   │   ├── output.mobi
│   │   └── output.pdf
│   ├── img
│   │   ├── contributors
│   │   │   └── judith-keller.jpg
│   │   ├── cover.png
│   │   ├── figures
│   │   │   ├── duo.jpg
│   │   │   ├── evans-burroughs.jpg
│   │   │   ├── evans-child.jpg
│   │   │   ├── evans-graveyard.jpg
│   │   │   ├── evans-hymns.jpg
│   │   │   ├── evans-legionnaire.jpg
│   │   │   ├── evans-refugee.jpg
│   │   │   ├── evans-sons.jpg
│   │   │   ├── evans-video-still.jpg
│   │   │   ├── evans-washroom.jpg
│   │   │   ├── evans.jpg
│   │   │   ├── family.jpg
│   │   │   ├── lange-house.jpg
│   │   │   ├── lange.jpg
│   │   │   ├── mother-b.jpg
│   │   │   ├── mother-c.jpg
│   │   │   ├── mother.jpg
│   │   │   └── pickers.jpg
│   │   ├── logo-getty.png
│   │   └── pattern-spiral-overlay.png
│   └── js
│       └── custom.js
└── themes
    └── default
        ├── CHANGELOG.md
        ├── LICENSE.md
        ├── README.md
        ├── archetypes
        │   └── default.md
        ├── cypress.json
        ├── layouts
        │   ├── 404.html
        │   ├── _default
        │   │   ├── baseof.html
        │   │   └── single.html
        │   ├── contents
        │   │   └── single.html
        │   ├── cover
        │   │   └── single.html
        │   ├── data
        │   │   └── search-index.html
        │   ├── entry
        │   │   └── single.html
        │   ├── essay
        │   │   └── single.html
        │   ├── partials
        │   │   ├── analytics.html
        │   │   ├── cite-this.html
        │   │   ├── contents-list
        │   │   │   ├── card-image.html
        │   │   │   ├── closing-tags.html
        │   │   │   ├── contents-list.html
        │   │   │   ├── file-dir.html
        │   │   │   ├── file-path.html
        │   │   │   ├── list-item-type.html
        │   │   │   ├── list-item-types
        │   │   │   │   ├── abstract.html
        │   │   │   │   ├── brief.html
        │   │   │   │   ├── default.html
        │   │   │   │   └── grid.html
        │   │   │   └── section-heading.html
        │   │   ├── contents-list.html
        │   │   ├── contributor-info.html
        │   │   ├── contributor-list.html
        │   │   ├── copyright.html
        │   │   ├── debug-contents.html
        │   │   ├── dublin-core.html
        │   │   ├── error-message.html
        │   │   ├── figure-caption.html
        │   │   ├── figure-fallback.html
        │   │   ├── figure-image.html
        │   │   ├── figure-label.html
        │   │   ├── figure-soundcloud.html
        │   │   ├── figure-table.html
        │   │   ├── figure-vimeo.html
        │   │   ├── figure-website.html
        │   │   ├── figure-youtube.html
        │   │   ├── figures
        │   │   │   ├── caption.html
        │   │   │   ├── label.html
        │   │   │   └── media-types
        │   │   │       ├── fallback.html
        │   │   │       ├── image.html
        │   │   │       ├── soundcloud.html
        │   │   │       ├── table.html
        │   │   │       ├── vimeo.html
        │   │   │       ├── website.html
        │   │   │       └── youtube.html
        │   │   ├── footer-buttons.html
        │   │   ├── head.html
        │   │   ├── icon.html
        │   │   ├── icons-cc.html
        │   │   ├── icons.html
        │   │   ├── json-ld.html
        │   │   ├── link-list-arrow.html
        │   │   ├── link-list.html
        │   │   ├── menu-header.html
        │   │   ├── menu.html
        │   │   ├── metadata.html
        │   │   ├── navbar.html
        │   │   ├── open-graph.html
        │   │   ├── page-abstract.html
        │   │   ├── page-bibliography.html
        │   │   ├── page-class.html
        │   │   ├── page-contributor.html
        │   │   ├── page-header.html
        │   │   ├── page-title.html
        │   │   ├── page-tombstone.html
        │   │   ├── pdf-info.html
        │   │   ├── pdf-title-pages.html
        │   │   ├── scripts.html
        │   │   ├── search.html
        │   │   ├── site-title.html
        │   │   ├── tests
        │   │   │   ├── check-node-modules.html
        │   │   │   ├── check-page-one-class.html
        │   │   │   ├── check-page-weights.html
        │   │   │   └── run.html
        │   │   └── twitter-card.html
        │   ├── shortcodes
        │   │   ├── q-bibliography.html
        │   │   ├── q-cite.html
        │   │   ├── q-class.html
        │   │   ├── q-contributor.html
        │   │   ├── q-contributors.html
        │   │   ├── q-figure-group.html
        │   │   ├── q-figure-new.html
        │   │   ├── q-figure-zoom.html
        │   │   └── q-figure.html
        │   ├── sitemap.xml
        │   └── splash
        │       └── single.html
        ├── package-lock.json
        ├── package.json
        ├── source
        │   ├── css
        │   │   ├── application.scss
        │   │   ├── bulma-components.scss
        │   │   ├── colors.scss
        │   │   ├── components
        │   │   │   ├── _all.scss
        │   │   │   ├── container.sass
        │   │   │   ├── quire-buttons.scss
        │   │   │   ├── quire-citation.scss
        │   │   │   ├── quire-contents-list.scss
        │   │   │   ├── quire-contributors.scss
        │   │   │   ├── quire-copyright.scss
        │   │   │   ├── quire-cover.scss
        │   │   │   ├── quire-deepzoom.scss
        │   │   │   ├── quire-entry.scss
        │   │   │   ├── quire-figure.scss
        │   │   │   ├── quire-map.scss
        │   │   │   ├── quire-menu.scss
        │   │   │   ├── quire-navbar.scss
        │   │   │   ├── quire-page.scss
        │   │   │   ├── quire-popup.scss
        │   │   │   └── quire-search.scss
        │   │   ├── epub.scss
        │   │   ├── fonts.scss
        │   │   ├── layout.scss
        │   │   ├── print.scss
        │   │   ├── utilities.scss
        │   │   └── variables.scss
        │   ├── fonts
        │   │   ├── ibm-plex-sans-condensed
        │   │   │   ├── ibmplexsanscondensed-bold.eot
        │   │   │   ├── ibmplexsanscondensed-bold.woff
        │   │   │   ├── ibmplexsanscondensed-bold.woff2
        │   │   │   ├── ibmplexsanscondensed-bolditalic.eot
        │   │   │   ├── ibmplexsanscondensed-bolditalic.woff
        │   │   │   ├── ibmplexsanscondensed-bolditalic.woff2
        │   │   │   ├── ibmplexsanscondensed-italic.eot
        │   │   │   ├── ibmplexsanscondensed-italic.woff
        │   │   │   ├── ibmplexsanscondensed-italic.woff2
        │   │   │   ├── ibmplexsanscondensed-regular.eot
        │   │   │   ├── ibmplexsanscondensed-regular.woff
        │   │   │   └── ibmplexsanscondensed-regular.woff2
        │   │   ├── noto-sans
        │   │   │   ├── notosans-bold.eot
        │   │   │   ├── notosans-bold.woff
        │   │   │   ├── notosans-bold.woff2
        │   │   │   ├── notosans-bolditalic.eot
        │   │   │   ├── notosans-bolditalic.woff
        │   │   │   ├── notosans-bolditalic.woff2
        │   │   │   ├── notosans-italic.eot
        │   │   │   ├── notosans-italic.woff
        │   │   │   ├── notosans-italic.woff2
        │   │   │   ├── notosans-regular.eot
        │   │   │   ├── notosans-regular.woff
        │   │   │   └── notosans-regular.woff2
        │   │   └── noto-serif
        │   │       ├── notoserif-bold.eot
        │   │       ├── notoserif-bold.woff
        │   │       ├── notoserif-bold.woff2
        │   │       ├── notoserif-bolditalic.eot
        │   │       ├── notoserif-bolditalic.woff
        │   │       ├── notoserif-bolditalic.woff2
        │   │       ├── notoserif-italic.eot
        │   │       ├── notoserif-italic.woff
        │   │       ├── notoserif-italic.woff2
        │   │       ├── notoserif-regular.eot
        │   │       ├── notoserif-regular.woff
        │   │       └── notoserif-regular.woff2
        │   ├── img
        │   │   ├── cover.png
        │   │   ├── fullscreen.png
        │   │   ├── icons
        │   │   │   ├── arrow_forward.png
        │   │   │   ├── baseline-headset-24px.png
        │   │   │   ├── baseline-videocam-24px.png
        │   │   │   ├── baseline-website-24px.png
        │   │   │   ├── cc-by-nc-nd.png
        │   │   │   ├── cc-by-nc-sa.png
        │   │   │   ├── cc-by-nc.png
        │   │   │   ├── cc-by-nd.png
        │   │   │   ├── cc-by-sa.png
        │   │   │   ├── cc-by.png
        │   │   │   ├── cc-zero.png
        │   │   │   ├── close.png
        │   │   │   ├── down-arrow.png
        │   │   │   ├── download.png
        │   │   │   ├── favicon.ico
        │   │   │   ├── home.png
        │   │   │   ├── left-arrow.png
        │   │   │   ├── link.png
        │   │   │   ├── nav.png
        │   │   │   ├── right-arrow.png
        │   │   │   ├── search.png
        │   │   │   └── start.png
        │   │   ├── layers-2x.png
        │   │   ├── layers.png
        │   │   ├── marker-icon.png
        │   │   ├── pattern-spiral-overlay.png
        │   │   └── promo-screenshots
        │   │       ├── classic-catalogue.png
        │   │       ├── classic-entry.png
        │   │       ├── classic-essay.png
        │   │       ├── modern-contents.png
        │   │       ├── modern-cover.png
        │   │       └── modern-essay.png
        │   └── js
        │       ├── application.js
        │       ├── deepzoom.js
        │       ├── epub.js
        │       ├── fullscreen.js
        │       ├── helper.js
        │       ├── map.js
        │       ├── navigation.js
        │       ├── popup.js
        │       ├── search.js
        │       └── soundcloud-api.js
        ├── static
        │   ├── css
        │   │   └── epub.css
        │   ├── fonts
        │   │   ├── MaterialIcons-Regular.eot
        │   │   ├── MaterialIcons-Regular.woff
        │   │   ├── MaterialIcons-Regular.woff2
        │   │   ├── ibmplexsanscondensed-bold.eot
        │   │   ├── ibmplexsanscondensed-bold.woff
        │   │   ├── ibmplexsanscondensed-bold.woff2
        │   │   ├── ibmplexsanscondensed-bolditalic.eot
        │   │   ├── ibmplexsanscondensed-bolditalic.woff
        │   │   ├── ibmplexsanscondensed-bolditalic.woff2
        │   │   ├── ibmplexsanscondensed-italic.eot
        │   │   ├── ibmplexsanscondensed-italic.woff
        │   │   ├── ibmplexsanscondensed-italic.woff2
        │   │   ├── ibmplexsanscondensed-regular.eot
        │   │   ├── ibmplexsanscondensed-regular.woff
        │   │   ├── ibmplexsanscondensed-regular.woff2
        │   │   ├── notosans-bold.eot
        │   │   ├── notosans-bold.woff
        │   │   ├── notosans-bold.woff2
        │   │   ├── notosans-bolditalic.eot
        │   │   ├── notosans-bolditalic.woff
        │   │   ├── notosans-bolditalic.woff2
        │   │   ├── notosans-italic.eot
        │   │   ├── notosans-italic.woff
        │   │   ├── notosans-italic.woff2
        │   │   ├── notosans-regular.eot
        │   │   ├── notosans-regular.woff
        │   │   ├── notosans-regular.woff2
        │   │   ├── notoserif-bold.eot
        │   │   ├── notoserif-bold.woff
        │   │   ├── notoserif-bold.woff2
        │   │   ├── notoserif-bolditalic.eot
        │   │   ├── notoserif-bolditalic.woff
        │   │   ├── notoserif-bolditalic.woff2
        │   │   ├── notoserif-italic.eot
        │   │   ├── notoserif-italic.woff
        │   │   ├── notoserif-italic.woff2
        │   │   ├── notoserif-regular.eot
        │   │   ├── notoserif-regular.woff
        │   │   └── notoserif-regular.woff2
        │   └── img
        │       ├── baseline-website-24px.png
        │       ├── cover.png
        │       ├── fullscreen.png
        │       ├── fullscreen@2x.png
        │       ├── icons
        │       │   ├── arrow_forward.png
        │       │   ├── baseline-headset-24px.png
        │       │   ├── baseline-videocam-24px.png
        │       │   ├── baseline-website-24px.png
        │       │   ├── cc-by-nc-nd.png
        │       │   ├── cc-by-nc-sa.png
        │       │   ├── cc-by-nc.png
        │       │   ├── cc-by-nd.png
        │       │   ├── cc-by-sa.png
        │       │   ├── cc-by.png
        │       │   ├── cc-zero.png
        │       │   ├── close.png
        │       │   ├── down-arrow.png
        │       │   ├── download.png
        │       │   ├── favicon.ico
        │       │   ├── home.png
        │       │   ├── left-arrow.png
        │       │   ├── link.png
        │       │   ├── nav.png
        │       │   ├── right-arrow.png
        │       │   ├── search.png
        │       │   ├── soundcloud.png
        │       │   ├── start.png
        │       │   ├── vimeo.png
        │       │   ├── website.png
        │       │   └── youtube.png
        │       ├── layers-2x.png
        │       ├── layers.png
        │       ├── marker-icon.png
        │       └── pattern-spiral-overlay.png
        ├── tests
        │   └── e2e
        │       ├── fixtures
        │       │   └── blns.json
        │       ├── plugins
        │       │   └── index.js
        │       ├── specs
        │       │   ├── quire-contents.spec.js
        │       │   ├── quire-contributors.spec.js
        │       │   ├── quire-entry.spec.js
        │       │   ├── quire-essay.spec.js
        │       │   ├── quire-menu.spec.js
        │       │   ├── quire-modal.spec.js
        │       │   ├── quire-next-back-buttons.spec.js
        │       │   ├── quire-page.spec.js
        │       │   ├── quire-search.spec.js
        │       │   └── quire-splash.spec.js
        │       └── support
        │           ├── commands.js
        │           └── index.js
        ├── theme.toml
        ├── tsconfig.json
        └── webpack
            ├── webpack.config.dev.js
            ├── webpack.config.epub.js
            ├── webpack.config.pdf.js
            └── webpack.config.prod.js

61 directories, 351 files
</pre>
</details>

Any advice on how I can proceed? Other commands like `quire site` also fail, since webpack isn't installed (since package.json and node_modules is missing).

**Comment by geealbers** [comment]
Hi @stamen-ag, try running `quire install` inside you're project directory. That should install the necessary `node_modules` inside the `themes/default/` directory. This installation should have been automatically processed when you ran `quire new` but it looks like for some reason it wasn't. Once the install process is complete, try `quire preview` again. If it's still not working, you can run `quire preview --verbose` which will output more logging and will hopefully point us to the source.

(Note that if you're running macOS Monterey, you'll also need to install Python 2: https://www.python.org/downloads/release/python-2718/. This is a common issue recently so I mention it here just in case. Both this and the `node_modules` install will be issue that are cleared up when we release the latest version of Quire later this fall.)

**Reply by Erin-Cecele** [reply]
Hi @stamen-ag, just wanted to see if this advice fixed your preview issue. Let us know!

**Reply by ncamachoh** [reply]
Hi. I'm currently having this same issue and running `quire install` inside my directory didn't fix it. I'm working on the same version on Windows.

events.js:377
      throw er; // Unhandled 'error' event
      ^

Error: spawn node_modules\.bin\webpack.cmd ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:274:19)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:280:12)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn node_modules\\.bin\\webpack.cmd',
  path: 'node_modules\\.bin\\webpack.cmd',
  spawnargs: [
    '--watch',
    '-d',
    '--progress',
    '--color',
    '--config',
    'webpack\\webpack.config.dev.js'
  ]
}

Any idea how to fix it? Thanks!

**Reply by geealbers** [reply]
@ncamachoh sorry to hear it. First confirm you have a supported version of node.js installed by running `node --version` it should be 14.18.1. If it's not, you should be able to install that version with https://nodejs.org/download/release/v14.18.1/node-v14.18.1-x64.msi.

Once you've confirmed node is at the correct version, you can try installing the project dependencies directly, without the `quire` command. Try the following four commands, one at a time. They will change into the default theme folder where the dependencies need to be installed; install them; change back into your main project directory; and then run preview.

```
cd themes/default
npm install
cd ../..
quire preview
```

If preview now works, you don't need to run any of these commands again in the future except `quire preview`.
